docs/rfc/block-registration.md

```diff
@@ -0,0 +1,383 @@
+
+Besides the dashicon or the source of the SVG element, the icon object can contain background and foreground colors, this colors will appear with the icon when they are applicable e.g.: in the inserter.
+
+**Note:** Custom SVG icons are automatically wrapped in the [wp.components.SVG](https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/svg/) component to add accessibility attributes (aria-hidden, role, and focusable).
+
+
+### Description
+
+* Type: `string`
+* Optional
+* Localized: Yes
+* Property: `description`
+
+```json
+ { "description": "Introduce new sections and organize content to help visitors" }
+```
+
+This is a short description for your block, which can be translated with our translation functions. This will be shown in the block inspector.
+
+### Keywords
+
+* Type: `string[]`
+* Optional
+* Localized: Yes
+* Property: `keywords`
+
+```json
+{ "keywords": [ "keyword1", "keyword2" ] }
+```
+
+Sometimes a block could have aliases that help users discover it while searching. For example, an image block could also want to be discovered by photo. You can do so by providing an array of terms (which can be translated). It is only allowed to add as much as three terms per block.
+
+### Attributes
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `attributes`
+
+```json
+{ 
+	"attributes": {
+		"cover": {
+			"type": "string",
+			"source": "attribute",
+			"selector": "img",
+			"attribute": "src"
+		},
+		"author": {
+			"type": "string",
+			"source": "html",
+			"selector": ".book-author"
+		}
+	}
+}
+```
+
+Attributes provide the structured data needs of a block. They can exist in different forms when they are serialized, but they are declared together under a common interface.
+
+See the [the attributes documentation](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-attributes/) for more details.
+
+### Edit
+
+* Type: `string` (`WPDefinedPropertyFile`)
+* Optional
+* Localized: No
+* Property: `edit`
+
+```json
+{ "edit": "my-block-edit.js" }
+```
+
+This property is a pointer to a JavaScript file containing the edit function of the block type. The edit function describes the structure of your block in the context of the editor. This represents what the editor will render when the block is used.
+
+See the [Edit and Save](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-edit-save/) documentation for more details.
+
+**Important Note:**
+
+The `WPDefinedPropertyFile` type described here is a subtype of string, where the value must represent an absolute or relative path to a file by which a dynamic value or values of a property can be interpreted.
+
+_TBD: Describe the resolution behavior by file extension, and consider further extension-specific subtypes (e.g. WPDefinedJSPropertyFile)._
+
+### Save
+
+* Type: `string` (`WPDefinedPropertyFile`)
+* Optional
```

Comment:
At the moment `save` is required. I opened https://github.com/WordPress/gutenberg/pull/14510 to make it optional.

Reply:
It's now optional.